app/controllers/analysis_controller.rb

```diff
@@ -0,0 +1,420 @@
+# Analysis controller
+require 'json'
+
+class AnalysisController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    verify_is_admin
+  end
+
+=begin Go through all of the death records in Nightingale searching for
+the patients who are within the requested age range and find their causes of death.
+Keep track of the number of instances of each cause of death and then send back
+an appropriate hash to the front end side.
+=end
+  def analyze_age_records
+    verify_is_admin
+    decedent_sex = JSON.parse(params[:sex])
+    decedent_age_range = JSON.parse(params[:age_range]).collect{|i| i.to_i}
+    age_keys = {0..9 => "Ages 0-9",
+                10..19 => "Ages 10-19",
+                20..29 => "Ages 20-29",
+                30..39 => "Ages 30-39",
+                40..49 => "Ages 40-49",
+                50..59 => "Ages 50-59",
+                60..69 => "Ages 60-69",
+                70..79 => "Ages 70-79",
+                80..89 => "Ages 80-89",
+                90..99 => "Ages 90-99",
+                100..120 => "Ages 100 and up"}
+    all_ages = []
+    partial_age_hash = {}
+=begin For each of the ages specified, place them in the partial_age_hash hash
+    which will be used later to lookup the appropriate value when creating the key for decedent_frequency.
+=end
+    decedent_age_range.each do |val|
+      if val < 100
+        (val..val + 9).each do |age|
```

Comment:
This is confusing to me, what does `val..val` do exactly in the case? Maybe this would be a bit more readable if you did something like `val..(val + 9)`?